Restrict package versions

gnn opened the pull request:
This should fix issues #343, #556, #641 and #669.

The changes are part of the CI branch and have been manually patched into the weekend runs for the last three weeks or so, so they should work. Now we only have to make sure that the restrictions are really enough to fix the mentioned issues.

## Before merging into `dev`, please make sure that:

- [x] The `CHANGELOG.rst` was updated.
- [x] New and adjusted code is formatted using `black` and `isort`.
- [x] The `Dataset`-version is updated when existing datasets are adjusted. 
- [x] The branch was merged into the `continuous-integration/run-everything-over-the-weekend-v2`-branch.
- [x] The workflow is running successfully in `Schleswig-Holstein` mode.
- [x] The workflow is running successfully in `Everything` mode.

nesnoj reviewed (approved):
Thank you @gnn for tackling those issues!
A fresh installation using py 3.8, pip 22.0.4 on CI v2 works like a charm :+1:
I started a full DAG run and will report back in the next days.

I added the PR checklist and modified the fix keywords in the description to allow auto-closing.

gnn commented:
> Thank you @gnn for tackling those issues!

Well, the solutions where lying around for a while, I just had to collect all the information on the error conditions to put them into the commit messages. But yeah. I'm also glad this is hopefully going to be fixed soon.

> A fresh installation using py 3.8, pip 22.0.4 on CI v2 works like a charm, I did not run into any of the reported problems +1 I started a full DAG run and will report back in the next days.

Cool. Fingers crossed.

> I added the PR checklist

I intentionally removed that one, because most of the things where either not applicable or checked by default. But seeing it now reminds me that I probably would've forgotten one or two things, so thanks for that. I adjusted some of the things though. Will run the changes by @ClaraBuettner to see if they can be incorporated permanently.

> and modified the fix keywords in the description to allow auto-closing.

I undid that to keep the issue text more compact and readable and because I prefer to auto-close the issues via the commit message of the merge commit.

gnn commented:
> Thank you @gnn for tackling those issues! A fresh installation using py 3.8, pip 22.0.4 on CI v2 works like a charm, I did not run into any of the reported problems +1 I started a full DAG run and will report back in the next days.
> 
> I added the PR checklist and modified the fix keywords in the description to allow auto-closing.

Quick question. I assume with "reported problems" you mean the ones reported in issue #641?
